File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/AuthContract.java
package org.mifos.mobilewallet.mifospay.auth;

import org.mifos.mobilewallet.mifospay.base.BasePresenter;
import org.mifos.mobilewallet.mifospay.base.BaseView;

/**
 * Created by naman on 16/6/17.
 */

/**
 * This specifies the contract between the view and the presenter.
 */
public interface AuthContract {

    interface LoginView extends BaseView<LoginPresenter> {

        void disableLoginButton();

        void enableLoginButton();

        void loginSuccess();

        void loginFail(String message);
    }

    interface LoginPresenter extends BasePresenter {

        void handleLoginButtonStatus(String usernameContent, String passwordContent);

        void loginUser(String username, String password);
    }
}

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/AuthContract.kt
package org.mifos.mobilewallet.mifospay.auth

import org.mifos.mobilewallet.mifospay.base.BasePresenter
import org.mifos.mobilewallet.mifospay.base.BaseView

/**
 * Created by naman on 16/6/17.
 */
/**
 * This specifies the contract between the view and the presenter.
 */
interface AuthContract {
    interface LoginView : BaseView<LoginPresenter?> {
        fun disableLoginButton()
        fun enableLoginButton()
        fun loginSuccess()
        fun loginFail(message: String?)
    }

    interface LoginPresenter : BasePresenter {
        fun handleLoginButtonStatus(usernameContent: String?, passwordContent: String?)
        fun loginUser(username: String?, password: String?)
    }
}